projectsmember/Login_User.aspx.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Recaptcha.Web.UI.Controls;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+namespace projectsmember
+{
+    public partial class Login_User : System.Web.UI.Page
+    {
+        SqlConnection LOGConn = new SqlConnection(ConfigurationManager.ConnectionStrings["sojaIrishConnectionString"].ConnectionString);
+
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            txtUserName.Attributes.Add("placeholder", "Username");
+            txtPass.Attributes.Add("placeholder", "Password");
+        }
+
+        protected void btnLoginUser_Click(object sender, EventArgs e)
+        {
+            if (String.IsNullOrEmpty(RecaptchaWidget3.Response))
+            {
+                lblMessage.Text = "Captcha cannot be empty.";
+            }
+            else
+            {
+                var result = RecaptchaWidget3.Verify();
+                if (result.Success)
+                {
+                    // Response.Redirect("Admin-Dashboard.aspx");
+                }
+                else
+                {
+                    lblMessage.Text = "رباتی؟ ";
+                    //foreach (var err in result.ErrorCodes)
+                    //{
+                    //   lblMessage.Text = lblMessage.Text + err;
+                    //}
+                }
+            }
+
+            DataSet1TableAdapters.MembersTableAdapter dstaMem = new DataSet1TableAdapters.MembersTableAdapter();
+            int numberOfUsers = dstaMem.QueryLoginUser(txtUserName.Text, txtPass.Text).GetHashCode();
+
+            if (numberOfUsers > 0)
+            {
+                //Login successful !
+                Session.Add("status", "Login_User");
```

Comment:
someone can just set the same session values and pretend that is already logged in. We need to do the followings here: 

if the user is successfully passed the username/password checking (authentication) then we generate a random string (token), and saved it somewhere in our database (maybe user table in an additional column for each user) and then return and save it in the session. 


then in the next pages, we say if what we have in session matches what we have in database (user doesn't need to do anything here) we assume the user is logged in. 

on each successful login, we generate a new random token ...

Reply:
Done, thanks for your help